lib/assert-result-factory.js

```diff
@@ -0,0 +1,33 @@
+/** @babel */
+
+class AssertResultFactory {
```

Comment:
I'm not saying the logic should be in TestRunnerProcess. It should be here, but it would be better to just export a simple function instead of a class with one function. Having it as a class here brings no benefit.